### DISCUSSION TITLE
Nvim-navic

### DISCUSSION THREAD

**Discussion by Caddy121** [LazyVim/LazyVim#104] [Q&A] [answered]
How do I move nvim-navic to the top of the buffer instead of it being in lualine?

**Comment by folke** [comment, marked as answer]
```lua
  -- lualine
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      -- dont show the winbar for some filetypes
      opts.options.disabled_filetypes.winbar = { "dashboard", "lazy", "alpha" }
      -- remove navic from the statusline
      local navic = table.remove(opts.sections.lualine_c)
      -- add it to the winbar instead
      opts.winbar = { lualine_b = { "filename" }, lualine_c = { navic } }
    end,
  },
}
```

Alternatively, you could disable just the lualine component, and add https://github.com/utilyre/barbecue.nvim

**Reply by Caddy121** [reply]
Awesome! Thank you for the solution!

**Comment by YangEfei** [comment]
Sorry to interrupt, but this question is perfect for my needs.
I want to remove the filetype and filename in the default config, too.
Is there any way except for 
```lua
table.remove(opts.sections.lualine_c, 2)
table.remove(opts.sections.lualine_c, 3)
```

Using indexes here does not seem very rigorous.
BTW, `local navic = table.remove(opts.sections.lualine_c)` is awesome.
How did it find navic? I don't seem to see a key named navic in the table.

**Reply by YangEfei** [reply]
I found the answer of how `local navic = table.remove(opts.sections.lualine_c)` works.
`table.remove(table, [,pos])`, if the pos is nil, it will remove the latest by default.
So interesting and reasonable.

**Reply by folke** [reply]
It's just a table, so you can do whatever you want with it.

You need to remove index 2 twice

```lua
table.remove(opts.sections.lualine_c, 2)
table.remove(opts.sections.lualine_c, 2)
```

**Reply by YangEfei** [reply]
Yup. Or execute `tabel.remove(opts.sections.lualine_c)` twice after remove navic is ok, too.
Thanks a lot for ur reply~

**Comment by marchank0** [comment]
Can't understand what I'm doing wrong:
1. Put this into `nvim/lua/config/lazy.lua` to activate default navic
```lua
 { import = "lazyvim.plugins.extras.editor.navic" },
```
2. Put this somewhere in `nvim/lua/plugins/somefile.lua` to customise conditional lualine configuration for navic:
```lua
  -- lualine
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      -- dont show the winbar for some filetypes
      opts.options.disabled_filetypes.winbar = { "dashboard", "lazy", "alpha" }
      -- remove navic from the statusline
      local navic = table.remove(opts.sections.lualine_c)
      -- add it to the winbar instead
      opts.winbar = { lualine_b = { "filename" }, lualine_c = { navic } }
    end,
  },
}
```

It doesn't work. Maybe I missed something important, but it seems like [this optional lualine config](https://www.lazyvim.org/extras/editor/navic#lualinenvim-optional) activates **after** my custom config and because of that I can't move navic to winbar

<img width="1439" alt="Screenshot 2024-03-31 at 13 28 48" src="https://github.com/LazyVim/LazyVim/assets/1183579/77b99803-cb06-4ada-b900-a6f3ee133924">

**Reply by marchank0** [reply]
I found what I missed
You need to place `{ import = "lazyvim.plugins.extras.editor.navic" },` **before** `{ import = "plugins" },` because order is **important**

**Comment by codeRSH** [comment]
Folke's solution worked for me to get the nvim-navic moved to winbar. 

But now I want to make the winbar 'persistent' i.e. to stop it from auto hiding when I switch buffer (eg. when I go to explorer or symbols buffer etc.) . Auto Hiding feature moves my code up and down which is quite distracting. 

any solutions/hints please?
